templates/pre-commit.protect-master

```diff
@@ -19,12 +20,25 @@ fi
 # Get the current branch name
 CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
+# Check if the current branch is `master` or `main`
+case "$CURRENT_BRANCH" in
+    master|main)
+        echo "FATAL ERROR: You cannot commit directly to the $CURRENT_BRANCH branch."
+        exit 1
+        ;;
+esac
+
+# Exit if it's a merge commit
+if [ "$(git rev-list --parents -n 1 HEAD | wc -w)" -gt 2 ]; then
```

Comment:
Shouldn't it be `-gt 1`?

Reply:
Oh, I see, it emits two shas for each parent, so this is right as you have it.  Don't mind me.

Reply:
Turns out this was wrong anyway, because we need to look at the commit in the middle of being made, so I changed it.